google-cloud-vision/src/test/java/com/google/cloud/vision/it/ITSystemTest.java

```diff
@@ -0,0 +1,915 @@
+    productSearchClient.addProductToProductSet(productToProductSetRequest);
+
+    /* create reference image */
+    ReferenceImage createReferenceImage =
+        ReferenceImage.newBuilder().setUri(SAMPLE_BUCKET + "product_search/shoes_1.jpg").build();
+    CreateReferenceImageRequest imageRequest =
+        CreateReferenceImageRequest.newBuilder()
+            .setParent(formatProductName)
+            .setReferenceImage(createReferenceImage)
+            .build();
+    referenceImage = productSearchClient.createReferenceImage(imageRequest);
+    referenceImageName = getName(referenceImage.getName());
+    formatReferenceImageName =
+        ReferenceImageName.of(PROJECT_ID, COMPUTE_REGION, productName, referenceImageName)
+            .toString();
+  }
+
+  @AfterClass
+  public static void tearDown() {
+
+    /* delete reference image */
+    DeleteReferenceImageRequest imageRequest =
+        DeleteReferenceImageRequest.newBuilder().setName(formatReferenceImageName).build();
+    productSearchClient.deleteReferenceImage(imageRequest);
+
+    /* remove product from product set */
+    RemoveProductFromProductSetRequest removeProductRequest =
+        RemoveProductFromProductSetRequest.newBuilder()
+            .setName(formatProductSetName)
+            .setProduct(formatProductName)
+            .build();
+    productSearchClient.removeProductFromProductSet(removeProductRequest);
+
+    /* delete product set */
+    DeleteProductSetRequest deleteProductSetRequest =
+        DeleteProductSetRequest.newBuilder().setName(formatProductSetName).build();
+    productSearchClient.deleteProductSet(deleteProductSetRequest);
+
+    /* delete product */
+    DeleteProductRequest deleteProductRequest =
+        DeleteProductRequest.newBuilder().setName(formatProductName).build();
+    productSearchClient.deleteProduct(deleteProductRequest);
+    productSearchClient.close();
+
+    imageAnnotatorClient.close();
+  }
+
+  @Test
+  public void detectFacesTest() throws IOException {
+    ByteString imgBytes =
+        ByteString.readFrom(new FileInputStream(RESOURCES + "face_no_surprise.jpg"));
+    Image img = Image.newBuilder().setContent(imgBytes).build();
+    Feature feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build();
+    AnnotateImageRequest request =
+        AnnotateImageRequest.newBuilder().addFeatures(feat).setImage(img).build();
+    BatchAnnotateImagesResponse response =
+        imageAnnotatorClient.batchAnnotateImages(ImmutableList.<AnnotateImageRequest>of(request));
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+    for (AnnotateImageResponse res : responses) {
+      for (FaceAnnotation annotation : res.getFaceAnnotationsList()) {
+        assertEquals(Likelihood.POSSIBLE, annotation.getAngerLikelihood());
+        assertEquals(Likelihood.POSSIBLE, annotation.getJoyLikelihood());
+        assertEquals(Likelihood.LIKELY, annotation.getSurpriseLikelihood());
+      }
+    }
+  }
+
```

Comment:
Overall test looks very good to me. 
One thing could be improved:

There are many tests staring with very similar code:

```
    ImageSource imgSource =
        ImageSource.newBuilder().setGcsImageUri(SAMPLE_BUCKET + "text/screen.jpg").build();
    Image img = Image.newBuilder().setSource(imgSource).build();
    Feature feat = Feature.newBuilder().setType(Type.DOCUMENT_TEXT_DETECTION).build();
    AnnotateImageRequest request =
        AnnotateImageRequest.newBuilder().addFeatures(feat).setImage(img).build();
    BatchAnnotateImagesResponse response =
        imageAnnotatorClient.batchAnnotateImages(ImmutableList.<AnnotateImageRequest>of(request));
    List<AnnotateImageResponse> responses = response.getResponsesList();

```
Would it make sense to introduce a function accepting several parameters and returning `List<AnnotateImageResponse>`?
Most of tests will become two times shorter?

Reply:
I tried you suggestion and run test cases couple of time but some times it works and some times not so keeping as it it.

Reply:
My approach doesn't suggest any changes in behavior. It's only about refactoring. So tests should work as previously.